File: 配套资源/配套代码/程序设计竞赛专题挑战教程-代码/第8章刷题统计.cpp
//刷题统计  2022年第十三届蓝桥杯省赛 C/C++大学B组
//lanqiaoOJ 2098


#include<bits/stdc++.h>
using namespace std;
typedef long long ll;
int main(){
    ll a,b,n; cin>>a>>b>>n;
    ll week = a*5+b*2;     //每周做题
    ll days = (n/week)*7;  //天数
    n %= week;             //余数
    if(n<=a*5) days += n/a+(n%a?1:0);  //在周一到周五内
    else{                  //周六和周日
        days += 5, n -= a*5;
        days += n/b+(n%b?1:0);
    }
    cout<<days;
    return 0;
}

File: 配套资源/配套代码/程序设计竞赛专题挑战教程-代码/第8章刷题统计.py
#刷题统计  2022年第十三届蓝桥杯省赛 C/C++大学B组
#lanqiaoOJ 2098

a,b,n = map(int,input().split())
week = a*5+b*2
days = (n//week)*7                     #注意是//不是/
n %= week      
if n <= a*5: days += n//a+(1 if n%a>0 else 0)   #三目运算
else:
    days += 5
    n -= a*5
    days += n//b+(1 if n%b>0 else 0)
print(days)
